delta-app/src/main/java/io/cdap/delta/store/RemoteHttpClient.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.delta.store;
+
+import com.google.common.base.Stopwatch;
+import com.google.common.io.ByteStreams;
+import com.google.common.io.CharStreams;
+import io.cdap.cdap.api.ServiceDiscoverer;
+import io.cdap.cdap.api.retry.RetryableException;
+import io.cdap.cdap.proto.id.NamespaceId;
+import io.cdap.delta.app.service.AssessmentService;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.io.Reader;
+import java.net.HttpURLConnection;
+import java.nio.charset.StandardCharsets;
+import java.util.concurrent.TimeUnit;
+import javax.ws.rs.HttpMethod;
+
+/**
+ * Class to handle Api calls to System Service
+ */
+public class RemoteHttpClient {
+  private static final long TIMEOUT_MILLIS = TimeUnit.SECONDS.toMillis(10);
+  private static final long RETRY_BASE_DELAY_MILLIS = 200L;
+  private static final long RETRY_MAX_DELAY_MILLIS = TimeUnit.SECONDS.toMillis(5);
+  private static final double RETRY_DELAY_MULTIPLIER = 1.2d;
+  private static final double RETRY_RANDOMIZE_FACTOR = 0.1d;
+
+  ServiceDiscoverer serviceDiscoverer;
+
+  public RemoteHttpClient(ServiceDiscoverer context) {
+    this.serviceDiscoverer = context;
+  }
+
+  public String retryableApiCall(String connectionUrl, String ...method) {
+    long delay = RETRY_BASE_DELAY_MILLIS;
+    double minMultiplier = RETRY_DELAY_MULTIPLIER - RETRY_DELAY_MULTIPLIER * RETRY_RANDOMIZE_FACTOR;
+    double maxMultiplier = RETRY_DELAY_MULTIPLIER + RETRY_DELAY_MULTIPLIER * RETRY_RANDOMIZE_FACTOR;
+    Stopwatch stopWatch = new Stopwatch().start();
+
+    try {
+      while (stopWatch.elapsedTime(TimeUnit.MILLISECONDS) < TIMEOUT_MILLIS) {
+        try {
+          if (method[0].equals(HttpMethod.GET)) {
+            return executeGetCall(getHttpUrlConnection(connectionUrl));
+          } else if (method[0].equals(HttpMethod.POST) || method[0].equals(HttpMethod.PUT)) {
+            return executeCallWithBody(getHttpUrlConnection(connectionUrl), method[0], method[1]);
```

Comment:
don't pass arguments as an array with special meaning in each array slot.

Make the method a separate argument of type HttpMethod and the request body as a `@Nullable String`